Str.regexp doesn't work well with utf-8 string

<!--
Welcome to OCaml's Issue tracker!

OCaml's developers use this tracker for bugs and feature requests only, rather
than user support.

If you have questions about *using* OCaml, please ask at
https://discuss.ocaml.org (more people read Discuss than this tracker, and
you'll get confirmation of whether you've really found a bug or need a new
feature).

If your error came from the OCaml package manager, opam, (messages beginning
`[ERROR] The compilation of ...`), please start at
https://github.com/ocaml/opam-repository/issues/new.

Some libraries and tools which used to be part of OCaml are now maintained
separately. Please post questions about Graphics, Num, camlp4, LablTk, CamlDBM
or OCamlTopWin on Discuss or on their respective issue trackers (see [README.adoc](https://github.com/ocaml/ocaml/blob/trunk/README.adoc#separately-maintained-components)
for a full list).
-->

There some bugs related to unicode characters and regexp, eg \b doesn't work with unicode characters. When I was trying to replicate word boundary using advanced unicode characters, it doesn't work well as well. E.g, I am trying to replace "тая" or "Тая" with "тэя" or "Тэя" respectively if they are the whole words only with such a code:
```
#load "str.cma"

let regexp = Str.regexp {|\(^\|[-( \*!\?.,:;'"…«]\)\(Т\|т\)ая\($\|[-) \*!\?.,:;'"…»]\)|}
let str = "запертая ."
let s = Str.global_replace regexp {|\1\2эя\3|} str 
let _ = Printf.printf "%s" s
```
The result of the code is "запертэя .", while it should keep the string as it was "запертая .". However, if I remove the character "…" from the regexp, so it is looks like:
`{|\(^\|[-( \*!\?.,:;'"«]\)\(Т\|т\)ая\($\|[-) \*!\?.,:;'"…»]\)|}`
the code works properly, not changing the string.

Can you fix it, please?

--- a/otherlibs/str/str.mli
+++ b/otherlibs/str/str.mli
@@ -18,6 +18,10 @@
 
 (** {1 Regular expressions} *)
 
+(**
+  The {!Str} library provides regular expressions on sequences of bytes.
+  It is, in general, unsuitable to match Unicode characters.
+*)
 
 type regexp
 (** The type of compiled regular expressions. *)

